File: solutions/go/nth-prime/1/nth_prime.go
package prime

import "errors"

var ErrNoZero = errors.New("there is no zeroth prime")

// Nth returns the nth prime number. An error must be returned if the nth prime number can't be calculated ('n' is equal or less than zero)
func Nth(n int) (prime int, err error) {
	if n == 0 {
		return prime, ErrNoZero
	}

	var primeCount int
	for i := 0; primeCount < n; i++ {
		for x := 2; x <= i; x++ {
			if x < i && i%x == 0 {
				break
			}

			if x == i {
				prime = i
				primeCount++
			}
		}
	}

	return
}

File: solutions/typescript/nth-prime/1/nth-prime.ts
export function nth(n: number): number {
  if (n === 0) throw new Error("Prime is not possible")

  let index = 0
  for (let num = 2; ; num++) {
    if (isPrime(num)) {
      index++
      if (index === n) {
        return num
      }
    }
  }
}

function isPrime(nbr: number): boolean {
  if (nbr === 2) {
    return true
  }
  if (nbr % 2 === 0) {
    return false
  }

  const limit = Math.sqrt(nbr)

  for (let x = 3; x <= limit; x += 2) {
    if (nbr % x === 0) {
      return false
    }
  }

  return true
}
